### PR TITLE
Add space_guid to Cloud Foundry bind_resource extension in profile.md

### PR DESCRIPTION
**Pull request by mattmcneeney**
Cloud Foundry sends an additional field in the `context` object when creating a service binding called `bind_resource.space_guid`, which indicates the Cloud Foundry space that the binding is associated with. This is not to be confused with the `space_guid` currently found in the `context` object, which indicates the space in which the service instance was/is going to be created.

### CODE REVIEW

**Comment by duglin** [comment]
Is there any reason you didn't add this to the real bind_resource object? It says:
```
Platforms MAY choose to add additional ones as needed
```

**Comment by mattmcneeney** [comment]
@duglin Mainly because that object already contains CF-specific fields that we will likely want to move out of that object in future versions. The `context` object is designed for platform-specific fields like this, and so we thought we should start setting this precedent. What do you think?

**Comment by duglin** [comment]
Are you thinking that the bind_resource object will go away? I actually don't think "app_guid" or "route" aren't too CF specific (maybe just a little, but tolerable).  Anyway, the text for for bind_resource says:
```
The bind_resource object contains Platform specific information related to the context in
which the service will be used.
```
so it sounds like a direct match to me. But, if there's confusion around when to use "bind_resource" and "context" we should probably make that clearer in the spec.

**Comment by mattmcneeney** [comment]
No I don't think `bind_resource` will go away. I'm happy to put the new field in the top-level `bind_resource` object and I agree that it sounds suitable, but we should discuss as a group where we want things like this to live moving forward. I've added this to the agenda for next week's call.

**Comment by mattmcneeney** [comment]
I guess it also depends on whether or not we want multiple extension points for Platform specific information, or just one with `context`. Food for thought...

File: profile.md [hunk 1]
```diff
@@ -142,17 +140,16 @@ Note that when both the originating identity HTTP Header and the Context
...
 
 The following properties are defined for usage within a Cloud Foundry
 deployment:
 
-- `organization_guid`<br>
-  The organization GUID as defined by the Cloud Foundry
-  specification/project. This property MUST be a non-empty string serialized
-  as follows:
+- `organization_guid`  
+  The GUID of the organization that a Service Instance is associated with.
```
Comment (duglin) [inline profile.md]: The text here wasn't instance specific before, it was more generic. For now this is probably ok but we need to be careful that if "context" is ever used on messages that are not necessarily related to instances then we'll need to revert this change.

File: spec.md [hunk 2]
```diff
@@ -1112,7 +1112,7 @@ might not be able to provide this information at the time of the binding
 request, therefore the `bind_resource` and its fields are OPTIONAL.
 
 Below are some common fields that MAY be used. Platforms MAY choose to add
-additional ones as needed.
+additional ones as needed (see [conventions](profile.md#bind-resource-object)).
```
Comment (duglin) [inline spec.md]: s/conventions/Bind Resource Object/

File: profile.md [hunk 3]
```diff
@@ -232,6 +229,37 @@ part of a Kubernetes API call:
...
+In the [Open Service Broker API specification](spec.md), requests to
+[create a Service Binding](spec.md#binding) can contain a `bind_resource`
+object in which Platforms MAY choose to add additional fields.
+
+#### Cloud Foundry Bind Resource Object
+
+The following properties are defined for usage within a Cloud Foundry
+deployment:
+
+- `space_guid`
```
Comment (duglin) [inline profile.md]: Is this required or optional for CF?  We should be clear.

Comment (mattmcneeney) [reply profile.md]: optional - will make this clearer

File: spec.md (hunk 2 above)
Comment (mattmcneeney) [reply spec.md]: done

File: profile.md [hunk 4]
```diff
@@ -232,6 +229,37 @@ part of a Kubernetes API call:
...
+  this property MUST be a non-empty string serialized as follows:
+  ```
+  "space_guid": "space-guid-here"
+  ```
+  For example:
+  ```
+  "space_guid": "15823972-c216-4ba5-9f3f-e75b0b891297"
+  ```
+
+The following example shows a `bind_resource` object that might appear as part
```
Comment (avade) [inline profile.md]: When "might" it appear?

Comment (mattmcneeney) [reply profile.md]: When a CF deployment supports it, and when a binding is being created to an application

Comment (avade) [reply profile.md]: I would suggest you replace might, with the exact example / CF deployment version supported from.

Is there any way for a client to know when to depend on it from the API header?

Comment (mattmcneeney) [reply profile.md]: Currently we just version this file, so broker authors will know it is supported when CF sends `2.14` as `x-broker-api-version`. That's probably enough right?

File: profile.md [hunk 5]
```diff
@@ -16,8 +16,10 @@ or REQUIRED level requirements defined herein.
 - [Originating Identity Header](#originating-identity-header)
 - [Context Object](#context-object)
   - [Context Object Properties](#context-object-properties)
-  - [Cloud Foundry](#cloud-foundry)
-  - [Kubernetes](#kubernetes)
+  - [Cloud Foundry Context Object](#cloud-foundry-context-object)
+  - [Kubernetes Context Object](#kubernetes-context-object)
+- [Bind Resource Object](#bind-resource-object)
+  - [Cloud Foundry Bind Resource Object](#cloud-foundry-bind-resource-object)
```
Comment (duglin) [inline profile.md]: I really like that you added the extra words ("Context Object", "Bind Resource Object") to these titles. Just makes it easier to track what section you're in 
👍

Comment (duglin) [reply profile.md]: Having said that, something is a bit off with this TOC. In some cases we list level 4 stuff but not level 3 stuff.  And I think the platform specific context object stuff might be one level too deep.  If you don't want to fix those in this PR that's fine, I'll do it in a follow-on one- just thought I mention it before I forget.

Comment (mattmcneeney) [reply profile.md]: Yeah I noticed that. I'll just fix it in here whilst it's open as has no LGTXs.

Comment (mattmcneeney) [reply profile.md]: OK, PR updated. Now all level 2 and 3 titles are in the TOC.

File: profile.md [hunk 6]
```diff
@@ -232,14 +233,45 @@ part of a Kubernetes API call:
...
+In the [Open Service Broker API specification](spec.md), requests to
+[create a Service Binding](spec.md#binding) can contain a `bind_resource`
+object in which Platforms MAY choose to add additional fields.
+
+### Cloud Foundry Bind Resource Object
+
+The following properties are defined for usage within a Cloud Foundry
+deployment:
+
+- `space_guid` (_optional_)
```
Comment (duglin) [inline profile.md]: I think the use of `_optional_` here is new and I'd prefer to not confuse people so perhaps we should just say:
`This OPTIONAL property is the GUID of a ....`

Comment (mattmcneeney) [reply profile.md]: Done!

**Comment by duglin** [comment]
LGTM

**Comment by avade** [comment]
We can address "might" in another PR / issue. 

LGTM

**Comment by angarg12** [comment]
LGTM

**Comment by duglin** [comment]
just one more needed

**Comment by fmui** [comment]
LGTM
